src/platform/linker/loadall.c

```diff
@@ -20,27 +20,53 @@
 #include <assert.h>
 #include <sys/mman.h>
 
+char *service_name = NULL;
+
+struct private_symtab{
+	char name[MAX_SYMB_LEN];
+	unsigned long vma_val;
+};
+
 unsigned long getsym(bfd *obj, char* symbol)
 {
         long storage_needed;
-        asymbol **symbol_table;
+        asymbol **symbol_table = NULL;
```

Comment:
This optimization disallows any freeing of symbols returned by this function. I assume the linker never does this. Is there no way for the linker just to free unused symbols?

Reply:
I'm not sure. However as @gparmer is rewriting the linker, I don't suppose implementing this is worth it.